crates/common_utils/src/pii.rs

```diff
@@ -87,26 +99,77 @@ where
     }
 }
 
-/// Email address
+/// Strategy for masking Email
 #[derive(Debug)]
-pub struct Email;
+pub struct EmailStrategy;
 
-impl<T> Strategy<T> for Email
+impl<T> Strategy<T> for EmailStrategy
 where
-    T: AsRef<str>,
+    T: AsRef<str> + std::fmt::Debug,
 {
     fn fmt(val: &T, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         let val_str: &str = val.as_ref();
-        let is_valid = validate_email(val_str);
 
-        if is_valid.is_err() {
-            return WithType::fmt(val, f);
+        match Email::from_str(val_str) {
+            Ok(_email) => match val_str.split_once('@') {
+                Some((a, b)) => write!(f, "{}@{}", "*".repeat(a.len()), b),
+                None => WithType::fmt(val, f),
+            },
+            Err(..) => WithType::fmt(val, f),
         }
```

Comment:
```suggestion
        match val_str.split_once('@') {
            Some((a, b)) => write!(f, "{}@{}", "*".repeat(a.len()), b),
            None => WithType::fmt(val, f),
        },
```

We won't need to perform the validation here when masking emails.